### PR TITLE
enable early data when specifying `--enable-all` option

### PR DESCRIPTION
**Pull request by miyazakh**
# Description

Enable early data when enabled all. but still disable it for `--enable-distro`

# Testing
$ ./configure --enable-all|grep Early
expects 
* Early Data:                 yes

$ ./configure --enable-distro|grep Early
expects 
* Early Data:                 no

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -585,6 +585,8 @@ then
     test "$enable_trusted_ca" = "" && enable_trusted_ca=yes
     test "$enable_session_ticket" = "" && enable_session_ticket=yes
 
+    test "$enable_earlydata" = "" && enable_earlydata=yes
```
Comment (dgarske) [inline configure.ac:588]: I am okay with early data on for `--enable-all`, but I don't think we should have it on for `--enable-distro`, which is used for Debian. @SparkiDev thoughts?

Comment (SparkiDev) [reply configure.ac:588]: I agree.
EarlyData is one of those features where customers could use it without understanding the security limitations.
Having said that, they need to use specific APIs to send and get the data.
I would prefer to have it manually turned on in the distribution.

Comment (miyazakh) [reply configure.ac:589]: Understood. I disable early data for `--enable-distro`

**Comment by SparkiDev** [comment]
Note that 'Early Data' lines can get split up as client and server output is mixed.
Try multiple runs to ensure that the longer strings are detected properly.

File: scripts/tls13.test [hunk 2]
```diff
@@ -245,60 +245,74 @@ fi
...
-    grep 'Session Ticket' "$client_out_file"
-    session_ticket=$?
-    early_data_cnt=`grep 'Early Data' "$server_out_file" | wc -l`
-    if [ $session_ticket -eq 0 -a $early_data_cnt -ne 4 ]; then
-        RESULT=1
-    fi
-    if [ $RESULT -ne 0 ]; then
-        echo -e "\n\nIssue with TLS v1.3 Early Data - session ticket"
+    counter=0
+    while [ "$counter" -lt 3 ]; do
```
Comment (SparkiDev) [inline scripts/tls13.test:249]: Don't need to test it multiple times.
I just remember having issue with greping the full string.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: scripts/tls13.test (hunk 2 above)
Comment (miyazakh) [reply scripts/tls13.test]: Reverted.

**Review by dgarske — APPROVED** [review]

**Comment by SparkiDev** [comment]
I tested the code with EarlyData and debug and didn't see any issues.

**Review by SparkiDev — APPROVED** [review]

**Comment by miyazakh** [comment]
Sorry for that. Might be due to grep count line changes that Sean noted.

**Pull request merged**